### PR TITLE
Add note to the README about not using Snap or Flatpak versions

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/fwupd/fwupd/issues/2543

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +11,20 @@ Additional information is available at the website: https://fwupd.org/
 
 The most up to date compilation instructions are available in the [Wiki](https://github.com/fwupd/fwupd/wiki/Compilation).
 
+**NOTE:** In most cases end users should never compile fwupd from scratch; it's a
+complicated project with dozens of dependencies (and as many configuration options)
+and there's just too many things that can go wrong.
+
+Users should just have fwupd installed and updated by their distro, hopefully
+by a packager who does understand all the config options.
```
Comment (superm1) [inline README.md:19]: no need to throw shade at packagers

File: README.md [hunk 2]
```diff
@@ -13,6 +11,20 @@ Additional information is available at the website: https://fwupd.org/
...
+The distribution will have also done some testing with how fwupd interacts with
+other software on your system, for instance using GNOME Software.
+
+Installing fwupd from the [Snap Store](https://snapcraft.io/fwupd) or from
```
Comment (superm1) [inline README.md:23]: I sorta think we should kick people here instead: https://github.com/fwupd/fwupd/wiki/fwupd-snap

File: README.md [hunk 3]
```diff
@@ -13,6 +11,20 @@ Additional information is available at the website: https://fwupd.org/
...
+[Flathub](https://flathub.org/) might be useful to update a specific device on
```
Comment (superm1) [inline README.md:24]: And here https://github.com/fwupd/fwupd/wiki/fwupd-flatpak

Comment (hughsie) [reply README.md:24]: Agree, fixed.

File: README.md (hunk 2 above)
Comment (hughsie) [reply README.md:23]: Agree, fixed.

File: README.md (hunk 1 above)
Comment (hughsie) [reply README.md:19]: hah, that was meant more with British sarcasm :) What about `Users should just have fwupd installed and updated by their distro, managed and tested by a trusted package maintainer."

Comment (superm1) [reply README.md:19]: I don't think bringing a trusted vs untrusted packager into the equation matters.  Most people see distributions as an amorphous blob that provides a Linux OS.

Comment (hughsie) [reply README.md:19]: Maybe `Users should just have fwupd installed and updated by their distro, managed and tested by the package maintainer.`?

Comment (superm1) [reply README.md:19]: That's better

Comment (hughsie) [reply README.md]: Okay, fixed.

**Review by superm1 — APPROVED** [review]

**Pull request merged**